lib/vector-core/src/lib.rs

```diff
@@ -24,6 +24,7 @@
 #[cfg(feature = "api")]
 pub mod api;
 pub mod config;
+pub mod enrichment_table;
```

Comment:
nit, maybe just call this module `enrichment`?

* `EnrichmentTable` → `enrichment::Table`
* `EnrichmentTables` → `enrichment::Database`